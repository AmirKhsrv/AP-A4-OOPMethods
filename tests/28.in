best_book
show_author_info 1818
best_book
show_sorted_shelf 53 want_to_read Autobiography
best_book
credit 36
credit 74
show_author_info 593
recommend_second_approach 101
show_sorted_shelf 74 want_to_read Fantasy
show_sorted_shelf 55 read Autobiography
best_book
recommend_first_approach 72
recommend_second_approach 90
credit 65
best_book
show_sorted_shelf 121 currently_reading Fantasy
show_sorted_shelf 89 read History
show_sorted_shelf 62 want_to_read Romance
recommend_second_approach 0
best_book
show_author_info 284
credit 129
best_book
best_reviewer
credit 87
show_author_info 1285
credit 128
best_book
credit 69
show_author_info 779
