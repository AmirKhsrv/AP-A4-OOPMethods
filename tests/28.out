id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 1818
Name: Norman Rush
Year of Birth: 1957
Place of Birth: Marshall
Member Since: December 9 2012
Genres: Fiction
Books:
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 162
Title: Damaged: Natalie's Story
Genre: Young Adult
Author: Talia Jager
***
id: 370
Title: Eternal Eden
Genre: Fantasy
Author: Nicole  Williams
***
id: 222
Title: Everything Is Illuminated
Genre: Fiction
Author: Jonathan Safran Foer
***
id: 343
Title: Knife Sworn
Genre: Fantasy
Author: Mazarkis Williams
***
id: 238
Title: Lamb: The Gospel According to Biff Christ's Childhood Pal
Genre: Humor
Author: Christopher Moore
***
id: 290
Title: Pleasures of a Notorious Gentleman
Genre: Romance
Author: Lorraine Heath
***
id: 78
Title: Seven Sons
Genre: Dark
Author: Lili St. Germain
***
id: 198
Title: The Algebraist
Genre: Science Fiction
Author: Iain M. Banks
***
id: 61
Title: The Bronze Horseman
Genre: Historical
Author: Paullina Simons
***
id: 244
Title: The DUFF: Designated Ugly Fat Friend
Genre: Young Adult
Author: Kody Keplinger
***
id: 171
Title: Wicked: The Life and Times of the Wicked Witch of the West
Genre: Fantasy
Author: Gregory Maguire
***
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
0.006943
0.014735
id: 593
Name: Earlene Fowler
Year of Birth: 1993
Place of Birth: Ghana
Member Since: June 17 2019
Genres: Thriller
Books:
id: 303
Title: The Good Luck Charm
Genre: Romance
Author: Helena Hunting
Reviews:
id: 99 Rating: 3 Likes: 8 Date: November 16 2018
id: 142 Rating: 4 Likes: 0 Date: January 20 2019
id: 949 Rating: 1 Likes: 1 Date: May 14 2017
id: 1209 Rating: 1 Likes: 0 Date: August 22 2013
id: 1350 Rating: 5 Likes: 2 Date: September 25 2018
id: 1595 Rating: 1 Likes: 0 Date: July 1 2013
id: 1747 Rating: 1 Likes: 0 Date: November 13 2018
id: 1846 Rating: 4 Likes: 0 Date: July 9 2016
id: 3401 Rating: 3 Likes: 0 Date: January 1 2016
id: 3889 Rating: 5 Likes: 0 Date: October 17 2017
id: 4887 Rating: 5 Likes: 3 Date: May 4 2018
id: 5127 Rating: 5 Likes: 0 Date: November 9 2010
id: 5574 Rating: 3 Likes: 0 Date: October 6 2014
id: 6137 Rating: 3 Likes: 0 Date: June 28 2017
id: 6480 Rating: 1 Likes: 1 Date: May 1 2011
id: 7559 Rating: 0 Likes: 7 Date: August 19 2017
id: 7913 Rating: 3 Likes: 0 Date: July 10 2019
id: 8048 Rating: 2 Likes: 0 Date: December 23 2018
id: 8160 Rating: 1 Likes: 0 Date: February 26 2017
id: 8310 Rating: 2 Likes: 0 Date: November 29 2019
id: 8626 Rating: 5 Likes: 0 Date: June 16 2014
id: 8683 Rating: 4 Likes: 0 Date: June 20 2013
id: 8858 Rating: 0 Likes: 0 Date: December 16 2015
id: 8862 Rating: 4 Likes: 0 Date: August 26 2018
id: 8903 Rating: 5 Likes: 0 Date: July 11 2018
id: 9182 Rating: 2 Likes: 0 Date: October 12 2018
id: 10427 Rating: 2 Likes: 0 Date: July 28 2018
id: 10432 Rating: 3 Likes: 0 Date: January 9 2018
id: 10490 Rating: 4 Likes: 0 Date: October 20 2017
id: 11047 Rating: 2 Likes: 24 Date: October 15 2010
id: 12102 Rating: 2 Likes: 3 Date: April 22 2018
id: 12147 Rating: 4 Likes: 2 Date: December 12 2019
id: 12633 Rating: 3 Likes: 0 Date: March 21 2016
id: 13302 Rating: 4 Likes: 0 Date: April 18 2015
id: 14111 Rating: 3 Likes: 1 Date: September 14 2016
id: 14698 Rating: 2 Likes: 0 Date: October 22 2019
id: 16634 Rating: 2 Likes: 0 Date: October 13 2018
id: 26
Title: Kings Rising
Genre: Fantasy
Author: C.S. Pacat
***
id: 160
Title: The Lies of Locke Lamora
Genre: Fantasy
Author: Scott Lynch
***
id: 406
Title: The Mating
Genre: Fantasy
Author: Nicky Charles
***
id: 327
Title: The Neverending Story
Genre: Fantasy
Author: Michael Ende
***
id: 202
Title: Charlotte's Web
Genre: Classics
Author: E.B. White
***
id: 126
Title: Diva
Genre: Historical
Author: Jillian Larkin
***
id: 174
Title: Five Days at Memorial: Life and Death in a Storm-Ravaged Hospital
Genre: Nonfiction
Author: Sheri  Fink
***
id: 346
Title: Genome: the Autobiography of a Species in 23 Chapters
Genre: Science
Author: Matt Ridley
***
id: 347
Title: I Thee Wed
Genre: Romance
Author: Celeste Bradley
***
id: 22
Title: The Children of Men
Genre: Fiction
Author: P.D. James
***
id: 136
Title: Washington's Crossing
Genre: History
Author: David Hackett Fischer
***
id: 375
Title: West of Here
Genre: Fiction
Author: Jonathan Evison
***
id: 28
Title: Charged
Genre: Romance
Author: Jay Crownover
***
id: 187
Title: Endure
Genre: Young Adult
Author: Carrie Jones
***
id: 146
Title: Forever Fae
Genre: Fantasy
Author: L.P. Dover
***
id: 293
Title: It's Kind of a Funny Story
Genre: Young Adult
Author: Ned Vizzini
***
id: 250
Title: It's the Stars Will Be Our Lamps
Genre: Cultural
Author: Alan Walsh
***
id: 176
Title: Keturah and Lord Death
Genre: Fantasy
Author: Martine Leavitt
***
id: 266
Title: Madhur Jaffrey's World Vegetarian: More Than 650 Meatless Recipes from Around the World
Genre: Food and Drink
Author: Madhur Jaffrey
***
id: 396
Title: Tao Te Ching
Genre: Philosophy
Author: Lao Tzu
***
id: 5
Title: The Secret Wisdom of the Earth
Genre: Fiction
Author: Christopher Scotton
***
id: 378
Title: The Tycoon's Revenge
Genre: Romance
Author: Melody Anne
***
id: 263
Title: Timepiece
Genre: Young Adult
Author: Myra McEntire
***
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 82
Title: Genghis Khan and the Making of the Modern World
Genre: History
Author: Jack Weatherford
Reviews:
id: 245 Rating: 1 Likes: 1103 Date: February 12 2011
id: 1978 Rating: 3 Likes: 0 Date: May 26 2009
id: 2419 Rating: 4 Likes: 0 Date: June 4 2018
id: 2503 Rating: 3 Likes: 0 Date: January 7 2019
id: 2563 Rating: 5 Likes: 0 Date: June 27 2016
id: 3171 Rating: 0 Likes: 0 Date: August 13 2015
id: 3338 Rating: 5 Likes: 0 Date: December 21 2017
id: 3556 Rating: 5 Likes: 0 Date: March 29 2018
id: 3774 Rating: 0 Likes: 0 Date: October 1 2011
id: 3883 Rating: 1 Likes: 0 Date: July 24 2019
id: 4155 Rating: 3 Likes: 3 Date: March 17 2017
id: 4772 Rating: 4 Likes: 0 Date: January 11 2018
id: 5291 Rating: 1 Likes: 0 Date: October 18 2019
id: 5481 Rating: 0 Likes: 0 Date: June 27 2019
id: 5674 Rating: 1 Likes: 0 Date: October 13 2018
id: 6473 Rating: 5 Likes: 4 Date: December 3 2014
id: 6568 Rating: 4 Likes: 3 Date: December 6 2018
id: 6572 Rating: 5 Likes: 6 Date: May 7 2015
id: 6611 Rating: 1 Likes: 2 Date: October 18 2018
id: 7436 Rating: 0 Likes: 2 Date: May 10 2017
id: 7886 Rating: 5 Likes: 0 Date: June 3 2018
id: 9102 Rating: 5 Likes: 0 Date: January 15 2014
id: 9385 Rating: 5 Likes: 0 Date: December 5 2019
id: 9645 Rating: 3 Likes: 0 Date: May 22 2019
id: 10158 Rating: 3 Likes: 0 Date: August 26 2014
id: 10628 Rating: 4 Likes: 1 Date: March 3 2019
id: 10638 Rating: 0 Likes: 0 Date: July 21 2012
id: 11073 Rating: 1 Likes: 0 Date: February 8 2018
id: 12096 Rating: 0 Likes: 1 Date: April 12 2018
id: 12761 Rating: 4 Likes: 0 Date: June 14 2019
id: 13298 Rating: 2 Likes: 0 Date: January 2 2018
id: 13434 Rating: 2 Likes: 0 Date: March 8 2017
id: 13684 Rating: 5 Likes: 0 Date: December 27 2019
id: 13932 Rating: 1 Likes: 0 Date: July 28 2014
id: 14519 Rating: 1 Likes: 0 Date: March 18 2016
id: 14653 Rating: 2 Likes: 1 Date: March 28 2013
id: 15057 Rating: 5 Likes: 0 Date: May 7 2017
id: 15171 Rating: 4 Likes: 0 Date: October 17 2011
id: 15763 Rating: 0 Likes: 1 Date: October 27 2018
id: 15815 Rating: 0 Likes: 0 Date: September 3 2017
id: 15968 Rating: 4 Likes: 2 Date: June 12 2018
id: 16595 Rating: 2 Likes: 0 Date: November 15 2018
id: 211
Title: The Fall of Five
Genre: Young Adult
Author: Pittacus Lore
Reviews:
id: 76 Rating: 5 Likes: 0 Date: August 19 2016
id: 490 Rating: 4 Likes: 0 Date: January 26 2018
id: 1006 Rating: 1 Likes: 3 Date: January 25 2017
id: 1861 Rating: 4 Likes: 0 Date: February 9 2018
id: 2686 Rating: 0 Likes: 0 Date: September 11 2016
id: 2727 Rating: 5 Likes: 0 Date: November 26 2015
id: 3688 Rating: 2 Likes: 0 Date: September 9 2017
id: 3734 Rating: 1 Likes: 0 Date: April 15 2012
id: 4304 Rating: 3 Likes: 26 Date: September 21 2011
id: 4478 Rating: 4 Likes: 1 Date: March 27 2019
id: 4593 Rating: 5 Likes: 0 Date: January 16 2015
id: 4677 Rating: 4 Likes: 1 Date: September 1 2016
id: 4857 Rating: 5 Likes: 4 Date: September 19 2017
id: 5397 Rating: 5 Likes: 0 Date: October 10 2018
id: 6087 Rating: 1 Likes: 0 Date: June 9 2011
id: 6101 Rating: 4 Likes: 0 Date: March 9 2019
id: 7230 Rating: 1 Likes: 10 Date: December 29 2017
id: 7660 Rating: 5 Likes: 3 Date: April 24 2018
id: 8166 Rating: 5 Likes: 0 Date: January 6 2015
id: 8699 Rating: 1 Likes: 0 Date: July 25 2019
id: 9124 Rating: 0 Likes: 0 Date: February 9 2016
id: 9144 Rating: 3 Likes: 0 Date: December 15 2017
id: 9730 Rating: 1 Likes: 0 Date: June 9 2011
id: 9892 Rating: 2 Likes: 0 Date: May 27 2017
id: 10053 Rating: 5 Likes: 0 Date: October 12 2018
id: 10413 Rating: 5 Likes: 0 Date: December 14 2018
id: 10697 Rating: 0 Likes: 22 Date: April 26 2017
id: 10906 Rating: 0 Likes: 0 Date: October 8 2014
id: 10993 Rating: 1 Likes: 0 Date: October 20 2018
id: 11111 Rating: 2 Likes: 3 Date: February 28 2016
id: 11330 Rating: 0 Likes: 26 Date: September 29 2014
id: 11761 Rating: 5 Likes: 0 Date: April 24 2015
id: 11958 Rating: 3 Likes: 0 Date: October 23 2014
id: 12401 Rating: 1 Likes: 0 Date: August 13 2015
id: 12622 Rating: 0 Likes: 0 Date: November 12 2019
id: 13928 Rating: 2 Likes: 0 Date: November 29 2019
id: 14234 Rating: 2 Likes: 2 Date: March 15 2019
id: 14242 Rating: 1 Likes: 1 Date: April 4 2019
id: 14568 Rating: 0 Likes: 0 Date: February 27 2018
id: 16016 Rating: 3 Likes: 0 Date: December 20 2015
id: 16289 Rating: 4 Likes: 0 Date: August 14 2015
id: 16570 Rating: 4 Likes: 0 Date: October 13 2018
id: 16628 Rating: 1 Likes: 0 Date: January 10 2016
id: 16765 Rating: 4 Likes: 0 Date: October 27 2018
0.005694
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 241
Title: Kuttiedathi and Other Stories
Genre: Short Stories
Author: M.T. Vasudevan Nair
***
id: 60
Title: Annie on My Mind
Genre: Lgbt
Author: Nancy Garden
***
id: 275
Title: Beauty from Pain
Genre: Romance
Author: Georgia Cates
***
id: 187
Title: Endure
Genre: Young Adult
Author: Carrie Jones
***
id: 142
Title: Halflings
Genre: Paranormal
Author: Heather Burch
***
id: 293
Title: It's Kind of a Funny Story
Genre: Young Adult
Author: Ned Vizzini
***
id: 127
Title: Legend
Genre: Young Adult
Author: Marie Lu
***
id: 403
Title: Magic Triumphs
Genre: Fantasy
Author: Ilona Andrews
***
id: 94
Title: Meridian
Genre: Young Adult
Author: Amber Kizer
***
id: 151
Title: Mist of Midnight
Genre: Historical
Author: Sandra Byrd
***
id: 21
Title: Sybil: The Classic True Story of a Woman Possessed by Sixteen Personalities
Genre: Nonfiction
Author: Flora Rheta Schreiber
***
id: 286
Title: Three Daves
Genre: Romance
Author: Nicki Elson
***
id: 181
Title: Tiger's Voyage
Genre: Fantasy
Author: Colleen Houck
***
id: 55
Title: Twelve Sharp
Genre: Mystery
Author: Janet Evanovich
***
id: 337
Title: Dominion of the Eth
Genre: Romance
Author: J.C. Owens
***
id: 236
Title: Slow Burn
Genre: Romance
Author: Sam B. Morgan
***
id: 227
Title: A Hymn Before Battle
Genre: Science Fiction
Author: John Ringo
***
id: 321
Title: Feed
Genre: Horror
Author: Mira Grant
***
id: 50
Title: Genghis Khan and the Making of the Modern World
Genre: History
Author: Jack Weatherford
***
id: 127
Title: Legend
Genre: Young Adult
Author: Marie Lu
***
id: 237
Title: North and South
Genre: Classics
Author: Elizabeth Gaskell
***
id: 360
Title: Steve Jobs
Genre: Biography
Author: Walter Isaacson
***
id: 369
Title: The Golden Compass
Genre: Fantasy
Author: Philip Pullman
***
id: 30
Title: The Tibetan Book of Living and Dying
Genre: Philosophy
Author: Sogyal Rinpoche
***
id: 365
Title: Their Eyes Were Watching God
Genre: Classics
Author: Zora Neale Hurston
***
id: 348
Title: Zigzag
Genre: Young Adult
Author: Ellen Wittlinger
***
id: 303
Title: The Good Luck Charm
Genre: Romance
Author: Helena Hunting
Reviews:
id: 99 Rating: 3 Likes: 8 Date: November 16 2018
id: 142 Rating: 4 Likes: 0 Date: January 20 2019
id: 949 Rating: 1 Likes: 1 Date: May 14 2017
id: 1209 Rating: 1 Likes: 0 Date: August 22 2013
id: 1350 Rating: 5 Likes: 2 Date: September 25 2018
id: 1595 Rating: 1 Likes: 0 Date: July 1 2013
id: 1747 Rating: 1 Likes: 0 Date: November 13 2018
id: 1846 Rating: 4 Likes: 0 Date: July 9 2016
id: 3401 Rating: 3 Likes: 0 Date: January 1 2016
id: 3889 Rating: 5 Likes: 0 Date: October 17 2017
id: 4887 Rating: 5 Likes: 3 Date: May 4 2018
id: 5127 Rating: 5 Likes: 0 Date: November 9 2010
id: 5574 Rating: 3 Likes: 0 Date: October 6 2014
id: 6137 Rating: 3 Likes: 0 Date: June 28 2017
id: 6480 Rating: 1 Likes: 1 Date: May 1 2011
id: 7559 Rating: 0 Likes: 7 Date: August 19 2017
id: 7913 Rating: 3 Likes: 0 Date: July 10 2019
id: 8048 Rating: 2 Likes: 0 Date: December 23 2018
id: 8160 Rating: 1 Likes: 0 Date: February 26 2017
id: 8310 Rating: 2 Likes: 0 Date: November 29 2019
id: 8626 Rating: 5 Likes: 0 Date: June 16 2014
id: 8683 Rating: 4 Likes: 0 Date: June 20 2013
id: 8858 Rating: 0 Likes: 0 Date: December 16 2015
id: 8862 Rating: 4 Likes: 0 Date: August 26 2018
id: 8903 Rating: 5 Likes: 0 Date: July 11 2018
id: 9182 Rating: 2 Likes: 0 Date: October 12 2018
id: 10427 Rating: 2 Likes: 0 Date: July 28 2018
id: 10432 Rating: 3 Likes: 0 Date: January 9 2018
id: 10490 Rating: 4 Likes: 0 Date: October 20 2017
id: 11047 Rating: 2 Likes: 24 Date: October 15 2010
id: 12102 Rating: 2 Likes: 3 Date: April 22 2018
id: 12147 Rating: 4 Likes: 2 Date: December 12 2019
id: 12633 Rating: 3 Likes: 0 Date: March 21 2016
id: 13302 Rating: 4 Likes: 0 Date: April 18 2015
id: 14111 Rating: 3 Likes: 1 Date: September 14 2016
id: 14698 Rating: 2 Likes: 0 Date: October 22 2019
id: 16634 Rating: 2 Likes: 0 Date: October 13 2018
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 284
Name: Mary Norton
Year of Birth: 1983
Place of Birth: Libya
Member Since: April 14 2015
Genres: Children S Books, Fiction
Books:
0.005178
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
0.007133
id: 1285
Name: Charlotte Perkins Gilman
Year of Birth: 1995
Place of Birth: Ecuador
Member Since: September 7 2016
Genres: Poetry, Science Fiction, Sociology
Books:
0.005319
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
0.007009
id: 779
Name: Angie Sage
Year of Birth: 1982
Place of Birth: Guinea
Member Since: November 8 2015
Genres: Young Adult
Books:
